lib/util/package-rules/current-version.ts

```diff
@@ -18,26 +18,29 @@ export class CurrentVersionMatcher extends Matcher {
     if (is.undefined(matchCurrentVersion)) {
       return null;
     }
-
-    if (is.nullOrUndefined(currentValue)) {
-      return false;
-    }
-
-    const isUnconstrainedValue = !!lockedVersion;
+    const isUnconstrainedValue =
+      !!lockedVersion && is.nullOrUndefined(currentValue);
     const version = allVersioning.get(versioning);
     const matchCurrentVersionStr = matchCurrentVersion.toString();
     const matchCurrentVersionPred = configRegexPredicate(
       matchCurrentVersionStr
     );
 
     if (matchCurrentVersionPred) {
-      return !(!isUnconstrainedValue && !matchCurrentVersionPred(currentValue));
+      const compareVersion = lockedVersion ?? currentVersion ?? currentValue;
+      return (
+        !is.nullOrUndefined(compareVersion) &&
+        matchCurrentVersionPred(compareVersion)
+      );
```

Comment:
`matchCurrentVersion` should try to match against the version first, and only fall back to value if no version is found